x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/ReservedFieldNames.java

```diff
@@ -261,6 +261,7 @@
             DataFrameAnalyticsConfig.SOURCE.getPreferredName(),
             DataFrameAnalyticsConfig.DEST.getPreferredName(),
             DataFrameAnalyticsConfig.ANALYSES.getPreferredName(),
+            DataFrameAnalyticsConfig.FIELDS.getPreferredName(),
```

Comment:
This made me think that `fields` is actually one of those that I'd worry for conflicts in the config index. It is possible we need to have `fields` in some other config where it is simply an array of strings. I wonder if we should prefix it with something to make it less possible to get a conflict. Any ideas? Perhaps `select_fields` or `selected_fields`? Or `_fields`? Or `df_fields`?

Reply:
@dimitris-athanasiou I was thinking of the same thing while writing this. I think a simple `df_fields` or `analyses_fields` may work best. 

`select_fields` implies only *selection*, but this entry allows *selection* (`includes`) and *rejection* (`excludes`). 

@droberts195 @sophiec20 what do y'all think?

Reply:
I went ahead and changed it to `analyses_fields`, that seems more explanatory and fits with the config motif.